src/main/java/com/openlattice/edm/PostgresEdmManager.java

```diff
@@ -88,35 +92,53 @@ public PostgresEdmManager( HikariDataSource hds, PostgresTableManager ptm, Hazel
                 }
         );
     }
+
     public Set<EntitySet> getAllLinkingEntitySetsForEntitySet( UUID entitySetId ) {
         return ImmutableSet.copyOf( entitySets
                 .values( Predicates.equal( EntitySetMapstore.LINKED_ENTITY_SET_INDEX, entitySetId ) ) );
     }
 
-    public Map<UUID, Set<UUID>> getLinkingIdsByEntitySetIds( Set<UUID> entitySetIds ) {
-        String query =
-                "SELECT " + ENTITY_SET_ID.getName() + ", array_agg(" + LINKING_ID.getName() + ") AS " + LINKING_ID
-                        .getName() +
-                        " FROM " + ENTITY_KEY_IDS.getName() +
-                        " WHERE " + LINKING_ID.getName() + " IS NOT NULL AND " + ENTITY_SET_ID.getName()
-                        + " = ANY( ? ) " +
-                        " GROUP BY " + ENTITY_SET_ID.getName();
+    /**
+     * Queries the linking ids belonging to the entities in the requested entity sets.
+     *
+     * @param entitySetIds Set of ids of regular entity sets.
+     * @return Set of linking ids mapped by the requested entity set ids.
+     */
+    public PostgresIterable<Pair<UUID, Set<UUID>>> getLinkingIdsByEntitySetIds( Set<UUID> entitySetIds ) {
+        final var entitySetPartitions = partitionManager.getEntitySetsPartitionsInfo( entitySetIds ).values().stream()
+                .flatMap( partitionInfo -> partitionInfo.component1().stream() )
```

Comment:
component1?

Reply:
Yeah, getPartitions() would be more readable. https://kotlinlang.org/docs/reference/multi-declarations.html